### PR TITLE
Ngscheckmate

### LINKED ISSUE #3083: new subworkflow: bam_ngscheckmate
I think it would be useful to have a subworkflow to run the NGSCheckMate tool on bam files.
This tool will try and determine whether samples come from the same individual, using a set of pre-determined SNPs that are variable across the population.
I personally find this to be invaluable for ensuring that samples that are supposed to be matched actually are, and vice versa.

The tool is built for human data, although you could generate your own bed file showing SNPs for other species.

### Is there an existing subworkflow for this?

- [X] I have searched for the existing subworkflow

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this subworkflow, I added myself to the `Assignees` to facilitate tracking who is working on the subworkflow

### PR DESCRIPTION
**Pull request by SPPearce**
## PR checklist
New subworkflow for taking bam files and running ngscheckmate

Closes #3083 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: subworkflows/nf-core/bam_ngscheckmate/main.nf [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    BCFTOOLS_MPILEUP (ch_bam_bed, ch_fasta.collect(), false)
+    ch_versions = ch_versions.mix(BCFTOOLS_MPILEUP.out.versions)
+
+    BCFTOOLS_MPILEUP
+    .out
+    .vcf
+    .map{it[1]}
+    .collect()
+    .view()
```
Comment (maxulysse) [inline subworkflows/nf-core/bam_ngscheckmate/main.nf:25]:
```suggestion
```

File: subworkflows/nf-core/bam_ngscheckmate/main.nf [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    take:
+    ch_bam              // channel: [ val(meta), bam ]
+    ch_bed              // channel: [ bed ]
+    ch_fasta            // channel: [ fasta ]
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    ch_bam_bed  = ch_bam.combine(ch_bed).view()
```
[the remark this answers is not in the document]
Comment (maxulysse) [reply subworkflows/nf-core/bam_ngscheckmate/main.nf:15]:
```suggestion
    ch_bam_bed  = ch_bam.combine(ch_bed)
```

File: subworkflows/nf-core/bam_ngscheckmate/main.nf [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    BCFTOOLS_MPILEUP (ch_bam_bed, ch_fasta.collect(), false)
+    ch_versions = ch_versions.mix(BCFTOOLS_MPILEUP.out.versions)
+
+    BCFTOOLS_MPILEUP
+    .out
+    .vcf
+    .map{it[1]}
```
[the remark this answers is not in the document]
Comment (SPPearce) [reply subworkflows/nf-core/bam_ngscheckmate/main.nf:23]: Do you mean being explicit with the map?  Something like:
```suggestion
    .map{meta, vcf -> vcf}
```

File: subworkflows/nf-core/bam_ngscheckmate/main.nf (hunk 2 above)
[the remark this answers is not in the document]
Comment (SPPearce) [reply subworkflows/nf-core/bam_ngscheckmate/main.nf]: Sorry, should remove debugging statements!

**Review by maxulysse — APPROVED** [review]

**Pull request merged**